py3status/modules/usbguard.py

```diff
@@ -200,182 +82,94 @@ def post_config_hook(self):
-        format_button_permanent = self.py3.safe_format(self.format_button_permanent)
-        self.py3.composite_update(
-            format_button_permanent, {"index": "button_permanent"}
-        )
+        self.devices = self._get_devices()
 
         usbguard_data = {
-            "device": len(device_data),
-            "format_button_filter": format_button_filter,
-            "format_button_permanent": format_button_permanent,
-            "format_device": format_device,
-            "format_button_device": format_button_device,
-            "format_device_list": format_device_list,
+            "device": len(self.devices),
+            "format_device": self._format_device(self.devices),
         }
 
-        response = {
-            "cached_until": self.py3.time_in(10),
+        return {
+            "cached_until": self.py3.CACHE_FOREVER,
             "full_text": self.py3.safe_format(self.format, usbguard_data),
+            "urgent": True,
         }
 
-        if self.allow_urgent:
-            response["urgent"] = True
-
-        self.device_data = device_data
-        return response
-
     def on_click(self, event):
-        index = event["index"]
-        button = event["button"]
-        if isinstance(index, int):
+        if not isinstance(event["index"], str):
             return
-        elif index == "button_device":
-            if button == self.button_next:
-                self.device_index += 1
-                self.device_index %= self.device_length + 1
-            elif button == self.button_previous:
-                self.device_index -= 1
-                self.device_index %= self.device_length + 1
-        elif index == "button_filter":
-            if button == self.button_next:
-                self.filter_index += 1
-                self.filter_index %= self.filter_length
-            elif button == self.button_previous:
-                self.filter_index -= 1
-                self.filter_index %= self.filter_length
-        elif index == "button_permanent":
-            self.permanent = not self.permanent
-        else:
-            device_id, policy_name = index.split("/")
-            device_id = int(device_id)
-
-            if self.init["notifications"]:
-                for device in self.device_data:
-                    if device["id"] == device_id:
-                        device["policy"] = policy_name
-                        self._notify_user(device)
-                        break
 
-            policy = self.init["target"][policy_name]
-            self.proxy.applyDevicePolicy("(uub)", device_id, policy, self.permanent)
-            if self.init["permanent"]:
-                self.permanent = False
-            self.device_index = 0
+        device_id, policy_name = event["index"].split("/")
+        device_id = int(device_id)
+        policy = self.init["target"][policy_name]
+        self.proxy.applyDevicePolicy("(uub)", device_id, policy, False)
```

Comment:
Cause we remove `notification`, I guess you can `int(device_id)` to kill a line. Just saying.

Reply:
True, done :)